clientconn.go

```diff
@@ -555,14 +555,11 @@ func (cc *ClientConn) GetState() connectivity.State {
 // Notice: This API is EXPERIMENTAL and may be changed or removed in a later
 // release.
 func (cc *ClientConn) Connect() {
-	if cc.GetState() == connectivity.Idle {
-		cc.mu.Lock()
-		for ac := range cc.conns {
-			// TODO: should this be a signal to the LB policy instead?
-			go ac.connect()
-		}
-		cc.mu.Unlock()
+	cc.mu.Lock()
+	if cc.balancerWrapper != nil {
+		cc.balancerWrapper.updateCh.Put(exitIdle{})
```

Comment:
Don't call the channel directly. ClientConn shouldn't know internal of balancerWrapper. Add a method to balancerWrapper.

And if that method returns a bool, you can keep `go ac.connect()` inside cc

```go
cc.mu.Lock()
defer cc.mu.Unlock()
if cc.GetState() != connectivity.Idle {
  return
}
if cc.balancerWrapper.exitIdle() { // This method will do type assertion
  return
}
// fallback
for ac := range ccb.cc.conns {
  go ac.connect()
}
```

Reply:
Moved the updateCh pushing into ccBalancerWrapper, but I don't think things can change so drastically here (follow-up offline).